marshal.go

```diff
@@ -106,7 +106,13 @@ func parsePort(s string) int {
 func (i *InstanceMetadata) UnmarshalJSON(b []byte) error {
 	i.Raw = b
 	// TODO(cq) could actually parse Raw here, and in a parallel UnmarshalXML as well.
-	return nil
+	var err error
+	var m map[string]interface{}
+	if err = json.Unmarshal(b, &m); err == nil {
+		i.parsed = m
```

Comment:
Should [field "parsed"](https://github.com/hudl/fargo/blob/master/struct.go#L116) be exported as "Parsed"?
If not, apart from round-tripping the parsed metadata, what's the good of parsing it if clients can't read it or write it?